### ISSUE TITLE
Error while turning on Settings Sync on Ubuntu Budgie

### ISSUE THREAD

**Issue by huacat1017** [microsoft/vscode#142581]
Issue Type: <b>Bug</b>

When turning on Settings sync, I saw the error below:
![screen](https://user-images.githubusercontent.com/57604590/153116572-e96168be-e34a-449a-b76d-6e50e0b9a8e5.png)
and checked the error log to get more info:
```
[2022-02-09 11:21:02.447] [settingssync] [info] Did reset the local sync state.
[2022-02-09 11:21:02.453] [settingssync] [error] TypeError: Cannot read property 'getRemoteUserDataSyncConfiguration' of undefined
    at r.getUserDataSyncConfiguration (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:174299)
    at r.getPreviews (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:173887)
    at vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:168407
    at E (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:18:11018)
    at r.preview (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:168398)
    at R._call (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:179766)
    at R.call (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:178456)
    at a.onPromise (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:20:4940)
    at a.onRawMessage (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:20:4299)
    at vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:20:3555
    at fire (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:16:1712)
    at MessagePort.J (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:14:52406)
[2022-02-09 11:21:02.466] [settingssync] [info] Did reset the local sync state.
```
Settings sync works well on my Windows devices, but I can't turn it on on my Ubuntu Budgie deivce.

VS Code version: Code 1.64.1 (d6ee99e4c045a6716e5c653d7da8e9ae6f5a8b03, 2022-02-07T17:17:47.733Z)
OS version: Linux x64 5.13.0-28-generic
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-3470 CPU @ 3.20GHz (4 x 3155)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 2, 2|
|Memory (System)|3.74GB (0.13GB free)|
|Process Argv|--unity-launch --crash-reporter-id 18f54a31-6e73-4a9e-8304-3e1e9abe8cf5|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|budgie-desktop|
|XDG_CURRENT_DESKTOP|Budgie:GNOME|
|XDG_SESSION_DESKTOP|budgie-desktop|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-language-pack-zh-hant|MS-|1.64.6

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30425749
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
vsc1dst:30433059
pythonvs932:30410667
wslgetstarted:30433507
vs360cf:30404996
vsrem710:30416614

```

</details>

**Comment by sandy081** [comment]
May I know if you have settings option disabled while turning on settings sync?

**Comment by huacat1017** [comment]
UPDATE: I can use the settings sync on my Linux.
But I found that strange:
1. I checked 6 options (what to sync) and it showed that error, but later I tried to only check "Settings "and it turned on succesfully.
2. Then I checked all options and all datas now can sync well on my computer.

**Comment by pkhodaveissi** [comment]
Same for me:

`[2022-02-11 11:21:17.814] [settingssync] [info] Did reset the local sync state.
[2022-02-11 11:21:24.859] [settingssync] [info] Did reset the local sync state.
[2022-02-11 11:21:24.863] [settingssync] [error] TypeError: Cannot read property 'getRemoteUserDataSyncConfiguration' of undefined
    at r.getUserDataSyncConfiguration (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:174299)
    at r.getPreviews (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:173887)
    at vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:168407
    at E (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:18:11018)
    at r.preview (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:168398)
    at R._call (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:179766)
    at R.call (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:76:178456)
    at a.onPromise (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:20:4940)
    at a.onRawMessage (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:20:4299)
    at vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:20:3555
    at fire (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:16:1712)
    at MessagePort.J (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:14:52406)
[2022-02-11 11:21:24.893] [settingssync] [info] Did reset the local sync state.`

**Comment by huacat1017** [comment]
@sandy081 The problem was on my Linux machine. I mentioned the logs above(no more logs).
My Windows devices don't have that problem, so I can't reproduce it.
What options do you mean? (Configuring what to sync? Or the global settings?)

**Comment by sandy081** [comment]
Yes, options are for configuring what to sync

**Comment by huacat1017** [comment]
Like I mentioned before (2nd comment).
I tried:
enable all options (failed)
disable all options except Settings (succeed)
The strange thing is :After enable Settings option successfully, I tried to configure and re-enable all options(without turning off) and Settings Sync turned on with all options!
So now I can't reproduce the issue on my Linux device(I haven't tried to re-install VSCode to reproduce the issue, but I will try tommorrow :) )

**Comment by diep89** [comment]
Same happened to me. Exactly same error stack on both Windows machines. Did as huacat1017 said and sync started working again.

**Comment by sandy081** [comment]
To verify:

- Turn on settings sync
- Uncheck `settings` option
- Make sure settings sync is turned on without errors

**Comment by huacat1017** [comment]
I also tested 1.64.0-insider and found that Windows devices had this problem, too.
Bug fixed on 1.65.0-insider, Thank you.
